src/DoubleTheDonation/Payment.php

```diff
@@ -14,13 +14,29 @@ class Payment {
 	 */
 	public function addPaymentMeta( $payment_id, $payment_data ) {
 
-		$companyID = isset( $_POST['doublethedonation_company_id'] ) ? give_clean( $_POST['doublethedonation_company_id'] ) : '';
+        // v3
+        if (isset($_POST['dtd'])) {
+            $companyID = give_clean($_POST['dtd']['company_id']);
+            $companyName = give_clean($_POST['dtd']['company_name']);
+            $companyEnteredText = give_clean($_POST['dtd']['entered_text']);
+        }
```

Comment:
How is this being triggered without the field scope, is it using an old hook or something?

My preference would be to keep the field scope so in the future we don't have to come back and separate out the v2 logic.

Reply:
Resolved https://github.com/impress-org/give-double-the-donation/pull/23/commits/529b8add5a52a358b82b122ee63f1a36023e0828

Reply:
It's using `give_insert_payment` hook.

Reply:
Ah okay, we want to move away from relying on that hook as it's considered legacy.  It also is not a reliable hook for v3 forms as it would be called when a donation is first created in pending state (before custom fields). 

For reference we are [dispatching that hook from a model action](https://github.com/impress-org/givewp/blob/dbf061969a1878fa2cc93a92af3d2b5164c5509c/src/Donations/ServiceProvider.php#L73) for backwards compatibility.

The idea for field scopes is to have a direct way of processing the field without having to rely on those specific hooks.

Reply:
Can you direct me to where we are sending this data to DTD?

Reply:
@jonwaldstein https://github.com/impress-org/give-double-the-donation/blob/529b8add5a52a358b82b122ee63f1a36023e0828/src/DoubleTheDonation/Payment.php#L62

Reply:
Yikes, so that logic is really tied to the v2 form processing/meta storage huh?

How would you feel about extracting that out to be used in the v3 field scope so it's not relying on the v2 form structure?  Ideally we would use action scheduler too but I know i'm blowing up this scope 😅 

![Screenshot 2024-12-03 at 12 15 22 PM](https://github.com/user-attachments/assets/21d5afab-c77b-499a-a1f9-1f6ab2f56341)

Reply:
What about V2 forms? If we extract everything into the field scope, how will the v2 forms save this data?

Reply:
We could leave this for v2 forms but bail out early if it recognizes data coming from v3 forms.

Reply:
I see you added the bailout [c5c7c80](https://github.com/impress-org/give-double-the-donation/pull/23/commits/c5c7c8077f37fefbae90a62e7b65eaf6021944d9) but we still need to update the field scope to include the api request right?

Reply:
@jonwaldstein resolved https://github.com/impress-org/give-double-the-donation/pull/23/commits/b3cbef8dde70d0174b6a485819ef2fae547c1bb5